native/src/main/java/io/ballerina/lib/aws/redshift/nativeimpl/ClientProcessor.java

```diff
@@ -47,29 +47,27 @@ public static Object createClient(BObject client, BMap<BString, Object> clientCo
         BString passwordVal = clientConfig.getStringValue(Constants.ClientConfiguration.PASSWORD);
         String password = passwordVal == null ? null : passwordVal.getValue();
         String datasourceName = null;
-        String requestGeneratedKeys = Constants.RequestGeneratedKeysValues.ALL;
 
         BMap options = clientConfig.getMapValue(Constants.ClientConfiguration.OPTIONS);
         BMap<BString, Object> properties = ValueCreator.createMapValue();
         Properties poolProperties = null;
 
         if (options != null) {
-            properties = options.getMapValue(Constants.ClientConfiguration.PROPERTIES);
-            BString dataSourceNamVal = options.getStringValue(Constants.ClientConfiguration.DATASOURCE_NAME);
-            datasourceName = dataSourceNamVal == null ? null : dataSourceNamVal.getValue();
-            BString requestGeneratedKeysVal = options.getStringValue(
-                    Constants.ClientConfiguration.REQUEST_GENERATED_KEYS);
-            requestGeneratedKeys = requestGeneratedKeysVal == null ?
-                    Constants.RequestGeneratedKeysValues.ALL : requestGeneratedKeysVal.getValue();
-            if (properties != null) {
-                for (Object propKey : properties.getKeys()) {
-                    if (propKey.toString().toLowerCase(Locale.ENGLISH).matches(Constants.CONNECT_TIMEOUT)) {
-                        poolProperties = new Properties();
-                        poolProperties.setProperty(Constants.POOL_CONNECTION_TIMEOUT,
-                                properties.getStringValue((BString) propKey).getValue());
+            if (options.getMapValue(Constants.ClientConfiguration.PROPERTIES) != null) {
+                properties = options.getMapValue(Constants.ClientConfiguration.PROPERTIES);
+                BString dataSourceNamVal = options.getStringValue(Constants.ClientConfiguration.DATASOURCE_NAME);
+                datasourceName = dataSourceNamVal == null ? null : dataSourceNamVal.getValue();
+                if (properties != null) {
+                    for (Object propKey : properties.getKeys()) {
+                        if (propKey.toString().toLowerCase(Locale.ENGLISH).matches(Constants.CONNECT_TIMEOUT)) {
```

Comment:
What's the purpose of this check? I mean what's the purpose of using `toLowerCase()` here?

Another thing, if the `properties` is not null, the `propKey` must be of type `BString` right?